src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/FhirDataStore.cs

```diff
@@ -211,6 +214,49 @@ public async Task HardDeleteAsync(ResourceKey key, CancellationToken cancellatio
             }
         }
 
+        public async Task<HttpStatusCode> UpsertExportJobAsync(ExportJobRecord jobRecord, CancellationToken cancellationToken = default)
+        {
+            EnsureArg.IsNotNull(jobRecord, nameof(jobRecord));
+
+            try
+            {
+                var result = await _documentClient.UpsertDocumentAsync(
+                _collectionUri,
+                jobRecord,
+                new RequestOptions() { PartitionKey = new PartitionKey(jobRecord.PartitionKey) },
+                disableAutomaticIdGeneration: true,
+                cancellationToken: cancellationToken);
+
+                return result.StatusCode;
+            }
+            catch (DocumentClientException dce)
+            {
+                if (dce.Error?.Message?.Contains(GetValue(HttpStatusCode.RequestEntityTooLarge), StringComparison.Ordinal) == true)
+                {
+                    throw new RequestRateExceededException(dce.RetryAfter);
+                }
+
+                throw;
+            }
+        }
+
+        public async Task<ExportJobRecord> GetExportJobAsync(string jobId, CancellationToken cancellationToken = default)
+        {
+            EnsureArg.IsNotNullOrEmpty(jobId);
+
```

Comment:
HttpStatus code is a little bit weird here. When we switch to SQL, the calls won't be over HTTP and it will be odd. It should return ExportJobRecord and maybe a wrapper object representing the outcome of the operation.

Reply:
We already have `UpsertOutcome` enum.

Reply:
Actually, maybe split the insert and update? For insert, we will need to check to make sure hash matches and all that stuff but that's not necessary for update.

Reply:
Removed the UpsertExportJobAsync and added CreateExportJobAsync and ReplaceExportJobAsync